Assets/Tests/InputSystem/CoreTests_Actions_Interactions.cs

```diff
@@ -191,111 +191,62 @@ public void Actions_CanPerformPressInteraction()
-        Assert.That(performedReceivedCalls, Is.Zero);
-        Assert.That(canceledReceivedCalls, Is.EqualTo(1));
-        Assert.That(canceledAction, Is.SameAs(action));
-        Assert.That(canceledControl, Is.SameAs(gamepad.buttonSouth));
-        Assert.That(action.phase, Is.EqualTo(InputActionPhase.Waiting));
+            Assert.That(trace, Started<HoldInteraction>(action, gamepad.buttonSouth, time: 10.5, value: 1.0));
+            Assert.That(action.ReadValue<float>(), Is.EqualTo(1));
+            Assert.That(action.phase, Is.EqualTo(InputActionPhase.Started));
 
-        canceledReceivedCalls = 0;
+            trace.Clear();
 
-        InputSystem.QueueStateEvent(gamepad, new GamepadState().WithButton(GamepadButton.South), 10.5);
-        InputSystem.Update();
+            // Let time pass but stay under hold time.
+            currentTime = 10.75;
+            InputSystem.Update();
 
-        Assert.That(startedReceivedCalls, Is.EqualTo(1));
-        Assert.That(performedReceivedCalls, Is.Zero);
-        Assert.That(canceledReceivedCalls, Is.Zero);
-        Assert.That(startedAction, Is.SameAs(action));
-        Assert.That(startedControl, Is.SameAs(gamepad.buttonSouth));
-        Assert.That(action.phase, Is.EqualTo(InputActionPhase.Started));
+            Assert.That(trace, Is.Empty);
 
-        startedReceivedCalls = 0;
+            // Now exceed hold time. Make sure action performs and *stays* performed.
+            currentTime = 11;
+            InputSystem.Update();
 
-        runtime.currentTime = 10.75 + timeOffset;
-        InputSystem.Update();
+            Assert.That(trace,
+                Performed<HoldInteraction>(action, gamepad.buttonSouth, time: 11, duration: 0.5, value: 1.0));
+            Assert.That(action.phase, Is.EqualTo(InputActionPhase.Performed));
+            Assert.That(action.ReadValue<float>(), Is.EqualTo(1));
 
-        Assert.That(startedReceivedCalls, Is.Zero);
-        Assert.That(performedReceivedCalls, Is.Zero);
-        Assert.That(canceledReceivedCalls, Is.Zero);
-        Assert.That(startedAction, Is.SameAs(action));
-        Assert.That(startedControl, Is.SameAs(gamepad.buttonSouth));
-        Assert.That(action.phase, Is.EqualTo(InputActionPhase.Started));
+            trace.Clear();
 
-        runtime.currentTime = 11 + timeOffset;
-        InputSystem.Update();
+            // Release button.
+            Release(gamepad.buttonSouth, time: 11.5);
 
-        Assert.That(startedReceivedCalls, Is.Zero);
-        Assert.That(performedReceivedCalls, Is.EqualTo(1));
-        Assert.That(canceledReceivedCalls, Is.Zero);
-        Assert.That(performedAction, Is.SameAs(action));
-        Assert.That(performedControl, Is.SameAs(gamepad.buttonSouth));
-        Assert.That(action.phase, Is.EqualTo(InputActionPhase.Waiting));
+            Assert.That(trace, Canceled<HoldInteraction>(action, gamepad.buttonSouth, time: 11.5, duration: 1, value: 0.0));
+        }
```

Comment:
This one looks bad but basically I just rewrote the previous test with the now available `InputActionTrace`+`ActionConstraint` method which makes writing this kind of stuff just a ton easier. Also added a few extra checks.